### PR TITLE
Decimal operators

### PR DESCRIPTION
**Pull request by pnowojski**

### CODE REVIEW

File: presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java [hunk 1]
```diff
@@ -771,26 +768,19 @@ else if (v2 != 0x5a86c47a) {
     private static boolean exceedsOrEqualTenToThirtyEight(Slice decimal)
     {
         // 10**38=
-        // v0=0(0),v1=160047680(98a22400),v2=1518781562(5a86c47a),v3=1262177448(4b3b4ca8)
+        // i0 = 0(0), i1 = 160047680(98a22400), i2 = 1518781562(5a86c47a), i3 = 1262177448(4b3b4ca8)
+        // l0 = 0x98a2240000000000l, l1 = 0x4b3b4ca85a86c47al
 
-        int v3 = getInt(decimal, 3);
-        if (v3 >= 0 && v3 < 0x4b3b4ca8) {
+        long l1 = getLong(decimal, 1);
```
Comment (sopel39) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java]: l1, l2 -> low, high?

**Comment by sopel39** [comment]
I think methods `overflows` needs to be modified so that for `PRECISION == 38` it should call `throwIfOverflows`. Cast operators use `overflow` to ensure that result value fits. Make sure casts operators work.

Other that that, LGTM

File: presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java (hunk 1 above)
Comment (pnowojski) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java]: fixed

**Comment by pnowojski** [comment]
merged into `feature_fast_decimal`

**Pull request closed without merging**